### PR TITLE
Remove all DEPRECATED_CODE_3 option and all code that was guarded by it

### PR DESCRIPTION
**Pull request by dalg24**
Next release will be Kokkos 4.3, about a year after version 4.0, so now is a good time to get rid of these.

This PR removes the `Kokkos_ENABLE_DEPRECATED_CODE_3` option and the corresponding `KOKKOS_ENABLE_DEPRECATED_CODE_3` macro.  All associated code is purged.

Deprecating features being removed include:
* `InitArguments`
* `KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_*` macros
* `Experimental::{clamp, min, max, minmax}`
* using declarations in the `Experimental:: namespace` for all math functions and math constants
* `{OpenMP,HPX}::partition_master`
* `MasterLock`

### CODE REVIEW

File: core/src/OpenMP/Kokkos_OpenMP.hpp [hunk 1]
```diff
@@ -166,14 +149,7 @@ class OpenMP {
 };
 
 inline int OpenMP::impl_thread_pool_rank() noexcept {
-  // FIXME_OPENMP Can we remove this when removing partition_master? It's only
-  // used in one partition_master test
```
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP.hpp:170]: Used in `UniqueToken::acquire()` as stated in commit message

**Comment by masterleinad** [comment]
```
__w/kokkos/kokkos/core/src/Kokkos_MathematicalFunctions.hpp:17: error: unterminated #ifndef
   17 | #ifndef KOKKOS_MATHEMATICAL_FUNCTIONS_HPP
      | 
```

**Review by Rombur — APPROVED** [review]

**Review by rgayatri23 — APPROVED** [review]

**Comment by masterleinad** [comment]
Did we check in with customers who were relying on some of these features?

**Comment by dalg24** [comment]
> Did we check in with customers who were relying on some of these features?

No

**Review by masterleinad — APPROVED** [review]

**Pull request merged**